lib/reverse_sentence.rb

```diff
@@ -1,6 +1,23 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: o(n)
+# Space complexity: o(n)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+
+ unless my_sentence == nil 
+    j = -1
+    strings = my_sentence.scan(/(\s*)(\w*'*\w+[!,.?]*)(\s*)/).flatten
```

Comment:
So this takes a string and creates an array of words... You were asked to do this in place.  Consider reversing each letter in the larger string once, so the string is reversed.  Then going through and reversing each word.